crates/datafusion_ext/src/functions.rs

```diff
@@ -393,13 +393,33 @@ impl FromFuncParamValue for i64 {
                 ScalarValue::UInt8(Some(v)) => Ok(v as i64),
                 ScalarValue::UInt16(Some(v)) => Ok(v as i64),
                 ScalarValue::UInt32(Some(v)) => Ok(v as i64),
-                ScalarValue::UInt64(Some(v)) => Ok(v as i64), // TODO: Handle overflow?
+                ScalarValue::UInt64(Some(v)) => {
+                    if v > std::i64::MAX as u64 {
+                        return Err(ExtensionError::InvalidParamValue {
+                            param: v.to_string(),
+                            expected: "int64",
+                        });
+                    }
+                    Ok(v as i64)
+                }
+                ScalarValue::Utf8(Some(v)) | ScalarValue::LargeUtf8(Some(v)) => v
+                    .parse::<i64>()
+                    .map_err(|_| ExtensionError::InvalidParamValue {
+                        param: v,
+                        expected: "integer",
+                    }),
                 other => Err(ExtensionError::InvalidParamValue {
                     param: other.to_string(),
                     expected: "integer",
                 }),
             },
-
+            FuncParamValue::Ident(v) => {
+                v.parse::<i64>()
+                    .map_err(|_| ExtensionError::InvalidParamValue {
+                        param: v,
+                        expected: "integer",
+                    })
+            }
```

Comment:
what's the case for this one? this seems really weird to me. `f("1")` where `"1"` is a column?